dartagnan/src/main/java/com/dat3m/dartagnan/encoding/PropertyEncoder.java

```diff
@@ -48,10 +51,35 @@ public class PropertyEncoder implements Encoder {
     private final AliasAnalysis alias;
     private final RelationAnalysis ra;
 
+    // We may want to make this configurable or just keep the option fixed.
+    private final boolean doWeakTracking = true;
```

Comment:
do you expect any performance difference between the two?

Reply:
Hard to tell. The weak tracking is technically a slightly more simple encoding in the case we check for multiple properties. I wouldn't be surprised if there is no difference in performance, but who knows :).